syntax = "proto3";

package detection;

// The object detection service definition.
service ObjectDetection {
  // Sends an image and receives detected objects.
  rpc DetectObjects(Image) returns (DetectionResponse);
}

// The request message containing the image data.
message Image {
  bytes image = 1; // Image data in bytes, preferably compressed (e.g., JPEG)
}

// The response message containing the detection results.
message DetectionResponse {
  repeated DetectedObject objects = 1;
}

// Information about a detected object.
message DetectedObject {
  int32 id = 1; // Object ID
  string label = 2; // Object label (e.g., "person", "car")
  float confidence = 3; // Confidence score
  // Bounding box [x_min, y_min, x_max, y_max], normalized to [0, 1]
  repeated float bbox = 4; 
}